Dockerfile

```diff
@@ -0,0 +1,30 @@
+# Multistage builds for reducing image size: https://mukul-mehta.in/til/reducing-container-size/
+
+# Builder image for installing dependencies
+FROM python:3.10-slim AS builder
+
+# To allow pip to run as root
+ENV PIP_ROOT_USER_ACTION=ignore
+
+# Create a virtual environment for installing packages
+RUN python -m venv /opt/venv
+ENV PATH="/opt/venv/bin:$PATH"
+
+# Install dependencies in the builder image
+COPY requirements.txt requirements.txt
+RUN pip install --upgrade pip
+RUN pip install -r requirements.txt
+
+# Image for the application
+FROM python:3.10-slim
+
+# Copy dependencies from the builder image
+COPY --from=builder /opt/venv /opt/venv
+ENV PATH="/opt/venv/bin:$PATH"
+
+# Copy source files
+WORKDIR /app
+COPY src/ .
+
+# Run the app
+CMD ["python", "main.py"]
```

Comment:
No Port exposed, would need one. 
Never run a prod app usingh direct python main.py. Use something like a uwgi gunicorn , take a look at wimp's dockerfile for help

Reply:
Hmm why should you not run a prod app using `python main.py`?

Reply:
The port is read from the environment variables. I thought it would be better to expose it through `docker run` instead.

Reply:
@amrav afaik gunicorn helps in managing python application by running multiple processes concurrently, which would improve the performance. I guess never is a wrong word but yes gunicorn should be preferred to run python applications atleast on production.

Reply:
@harshkhandeparkar we would still need it in the dockerfile and nevertheless would need a docker-compose too.

Reply:
> we would still need it in the dockerfile

How can I add a dynamic port in the Dockerfile?

Reply:
I can expose the default 8080 port though

Reply:
Didnt see that this is a tornado http server, seems the use of gunicorn wont be needed.